src/ecdsa/keygen.rs

```diff
@@ -695,6 +699,31 @@ impl State<KeyGeneratorTraits> for Phase2 {
         }]))
     }
 }
+/// Computes the sum of points on the curve and validates every point
+/// Uses PublicKey type from underlying secp256k1 library to work around limited API in curv crate
```

Comment:
I disagree the curv API is limited  here - 
have you actually tried to use invalid point with the existing curv API ? 
whenever you construct a GE there is a check (done in the secp256k1 library in this case) for valid ec point. If I am not mistaken the error should be handled and propagate

Reply:
The func from_bytes() does that indeed. Traced down to C functions. Did not look into the support for other curves. 
The lib is good and useful, but 
* There is no dedicated point validation function
*  There are quite few unwrap() calls in the code. For example,  from_coor() employs unwrap() at the very end. 

 I guess line 703 above should be shorter anyway.
 Thanks!

Reply:
Thanks @RustMania , 

1. no dedicated point validation : it is by design - we wanted to use the built in point validation from the libraries curv is consuming. curv is supposed to be higher level library - kind of generic group model level :) 
2. agree, there is much room for improvement

(cc: @survived )